### DISCUSSION TITLE
custom autovacuum settings at table level for new partitions

### DISCUSSION THREAD

**Discussion by tahseenbaig** [pgpartman/pg_partman#625] [Q&A] [answered]
I need to change `autovacuum_vacuum_insert_scale_factor` at table level on one of the `INSERT` heavy tables that does not get hit by `UPDATE`s and `DELETE`s. However that table is partitioned and I can only run the `ALTER` commands on individual partitions. 

How can I ensure the custom setting for `autovacuum_vacuum_insert_scale_factor ` gets applied to the new partitions that `run_maintenance` creates? 
Is this already possible now and I missing something?

Thanks!

**Comment by keithf4** [comment, marked as answer]
As of pg_partman version 4.5.1, you can apply these settings to the template table and it should then apply to the child tables. Last I'd checked this was in PG15 and it wasn't working automatically from the parent, so the template is the only way to do this. I haven't checked PG16 yet.

I also just realized that I'm missing this inheritance feature from the documentation, so I'll go and get that added. Thanks for asking!

**Comment by keithf4** [comment]
Note that you will initially have to manually apply this to all existing children. But it will then apply automatically to any newly created children once the template is up to date.
